ocean_provider/utils/url.py

```diff
@@ -20,8 +20,26 @@
 CHUNK_SIZE = 8192
 
 
-def is_safe_url(url):
+def get_redirect(url):
     if not is_url(url):
+        return None
+
+    result = requests.head(url, allow_redirects=False)
+
+    if result.status_code == 405:
+        # HEAD not allowed, so defaulting to get
+        result = requests.get(url, allow_redirects=False)
+
+    if result.is_redirect:
+        return get_redirect(result.headers["Location"])
```

Comment:
Redirect may be local, and not a complete URL.

e.g. `Location: /internal-redirect.html`

Reply:
I've added local redirects + tests.